ui/app/components/transformation-edit.js

```diff
@@ -64,23 +46,39 @@ export default TransformBase.extend({
           errorStatus: e.httpStatus,
         };
       });
+    // if an error occurs while querying the role, exit function and return the error
+    if (roleStore.errorStatus) return roleStore;
+    // otherwise update the role with the transformation and save
+    let transformations = roleStore.transformations;
+    if (role.action === 'ADD') {
+      transformations = addToList(transformations, transformationId);
+    } else if (role.action === 'REMOVE') {
+      transformations = removeFromList(transformations, transformationId);
+    }
+    roleStore.setProperties({
+      backend,
+      transformations,
+    });
+    return roleStore.save().catch((e) => {
```

Comment:
It feels strange to have two catch blocks here - why did you originally remove the `.then`?

Reply:
Good question.

We returned the results from a .then() which returns a [new Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then). This was the main gist of the problem because this method gets called ([here](https://github.com/hashicorp/vault/blob/main/ui/app/components/transformation-edit.js#L72) is a link to the original code) on row 75. We then used then created a new array of new Promises, used the `all` to resolve all of them and moved on to error handling—but that was the gist of the error. Using unresolved promises. 

Using await returns a resolved promise, solving the problem. From there I went back and forth on how much to restructure this method based on that change. I opted to keep as much of the old structure as possible to avoid any regressions (and also because I thought it might be easier to read my changes in a review 🙃). That being said, if you think it no longer reads well I can revisit and make the new flow match more of async/await pattern instead of a patched async/await pattern. I could also maybe approach with a finally—something I didn't try.